pkg/topology/full/full.go

```diff
@@ -149,6 +159,60 @@ func (d *driver) ClosestPeer(addr swarm.Address) (swarm.Address, error) {
 	return closest, nil
 }
 
+func (d *driver) Close() error {
+	close(d.quit)
+	return nil
+}
+
+func (d *driver) backoff(tryAfter time.Time) {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+	if d.backoffActive {
+		return
+	}
+
+	d.backoffActive = true
+
+	done := make(chan struct{})
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		select {
+		case <-done:
+		case <-d.quit:
+			cancel()
```

Comment:
Cancel should be called always to release context resources, so I am suggesting to move `cancel()` call at the very end of this anon function after the select is passed by either `done` or `d.quit`. Another approach would be maybe a bit more idiomatic, to have a `defer cancel()` just after the cancel function is declared, as defer is now not expensive and it is allowed to call cancel multiple times.

Reply:
Hmm, I agree yes, as `AddPeer` can be async. Nice catch